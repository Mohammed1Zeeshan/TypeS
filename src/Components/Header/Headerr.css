img
{
    height: 10vmin;
    width: 15%;
    align-items:center;  
}
.nav
{
    background-color: rgb(39, 139, 167);
    color: rgb(83, 170, 170);
  
}
.c
{
    margin-left: 5cm;
    padding-top: 5mm;
    color: rgb(134, 121, 112);
    height: 10px;
    width: 50%; 
    text-align: center; 
}
.a
{
   margin-left:15cm;
    padding: 1px;;
    color: rgb(15, 83, 61);
}   
.b
{
  
    padding: 1px;
    column-fill: auto;
    color: rgb(37, 65, 55);

}
