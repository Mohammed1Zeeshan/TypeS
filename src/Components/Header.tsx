import React from 'react'
import { Container, Nav, Navbar} from 'react-bootstrap';
import { Link } from 'react-router-dom';
import logo from './image/logo.png';
import './Headerr.css';



function Header() {
  return (
    <div >
       <>
       <Navbar bg='dark' variant='dark'>
           
      
      <Nav >
       
          <img src={logo}/>
        <h3 className='c'> Welcome to Augmento labs</h3> 
        <h1 className='a'><Nav.Link as={Link} to='/'>Home</Nav.Link></h1> 
       <h1 className='b'> <Nav.Link as={Link} to='/about'>About</Nav.Link></h1>
       <h1 className='b'> <Nav.Link as={Link} to='/Adduser'>Adduser</Nav.Link></h1> 
      </Nav>
      
      </Navbar>
      </>
    </div>
    
  )
}

export default Header