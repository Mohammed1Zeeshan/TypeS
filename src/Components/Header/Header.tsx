import React from 'react'
import { Container, Nav, Navbar} from 'react-bootstrap';
import { Link } from 'react-router-dom';
import logo from './../image/logo.png';
import './Headerr.css';



function Header() {
  return (
    <div className='Head'>
       <>
       <Navbar bg='dark' variant='dark'>
           
      
      <Nav >
       
          <img className='logo' src={logo}/>
        <h3 className='c'> Welcome to Augmento labs</h3> 
        <h1 className='a'><Nav.Link as={Link} to='/home'>Home</Nav.Link></h1> 
       <h1 className='b'> <Nav.Link as={Link} to='/about'>About</Nav.Link></h1>
       <h1 className='b'> <Nav.Link as={Link} to='/Adduser'>Adduser</Nav.Link></h1> 
      </Nav>
      
      </Navbar>
      </>
    </div>
    
  )
}

export default Header