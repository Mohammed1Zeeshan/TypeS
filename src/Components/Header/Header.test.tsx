import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";
import  puppeteer  from "puppeteer-core";
import { toMatchImageSnapshot } from "jest-image-snapshot";
import { render, screen } from "@testing-library/react";

jest.setTimeout(30000)
expect.extend({toMatchImageSnapshot});
const browserPath="C:\\Program Files (x86)\\Google\\Chrome\\Application\\chrome.exe";


it('test',()=>{
    const Wrapper=shallow(<Header/>)
    expect(Wrapper.exists('.Head')).toMatchSnapshot();
})
// Image snapshot testing  not required its just for practise

it("image snapshot testing header",async()=>{
    const browser=await puppeteer.launch({executablePath:browserPath})
    const page=await browser.newPage();
    await page.goto('http://localhost:3006');
    const image= await page.screenshot()
    expect(image).toMatchImageSnapshot();
})
it('Home page nav button',()=>{
    const Wrapper=shallow(<Header/>)
    expect(Wrapper.exists('.a')).toMatchSnapshot();
})
it('About page nav and AddUser page nav button',()=>{
    const Wrapper=shallow(<Header/>)
    expect(Wrapper.exists('.b')).toMatchSnapshot();
})
it('About page nav and AddUser page nav button',()=>{
    const Wrapper=shallow(<Header/>)
    expect(Wrapper.exists('.logo')).toMatchSnapshot();
})



