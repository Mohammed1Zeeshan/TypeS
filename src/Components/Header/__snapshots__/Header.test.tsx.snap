// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page nav and AddUser page nav button 1`] = `true`;

exports[`About page nav and AddUser page nav button 2`] = `true`;

exports[`Home page nav button 1`] = `true`;

exports[`test 1`] = `true`;
